### PR TITLE
Added alternate docker commands for all python commands

### LINKED ISSUE #236: Modify all Wiki pages by adding an alternative Docker command for each Python command.
All commands in the wiki pages must be updated to have an alternative docker command
Format:
Python:
```
cmd
```
Docker:
```
cmd
```

### PR DESCRIPTION
**Pull request by jaiakash**
Since we are also having docker setup now, i have added alternative docker commands for the python commands. This PR adds the alternative docker commands. 

References - https://github.com/oppia/oppia/wiki/Installing-Oppia-using-Docker#running-tests

Fixes #236

### CODE REVIEW

**Comment by jaiakash** [comment]
Hi @gp201 This is the way i have added the alternative docker commands as suggested in the #236 . Please check it if its fine, I will update the other pages in the same branch once you give confirmation.

File: Acceptance-Tests.md [hunk 1]
```diff
@@ -46,11 +46,19 @@ Files included inside this directory are :
 
 ## How to run the acceptance tests
 From the root directory of oppia, run the following command:
+Python:
 `python -m scripts.run_acceptance_tests --suite={{suiteName}}`
 
+Docker:
+`make run_tests.acceptance suite=SUITE_NAME`
```
Comment (gp201) [inline Acceptance-Tests.md]:
``````suggestion
Docker:
```
make run_tests.acceptance suite=SUITE_NAME
```
``````

File: Acceptance-Tests.md [hunk 2]
```diff
@@ -46,11 +46,19 @@ Files included inside this directory are :
 
 ## How to run the acceptance tests
 From the root directory of oppia, run the following command:
+Python:
 `python -m scripts.run_acceptance_tests --suite={{suiteName}}`
```
Comment (gp201) [inline Acceptance-Tests.md]:
``````suggestion
Python:
```
python -m scripts.run_acceptance_tests --suite={{suiteName}}
```
``````

File: Webpack.md [hunk 3]
```diff
@@ -151,8 +151,13 @@ output: {
 Here we specify the build mode i.e. production and the output config. We don't use a devtool here to make the build faster.
 
 This is the config that is used while running the dev server using
+
+Python:
 `python -m scripts.start --prod_env`.
 
+Docker:
+`make run-devserver prod_env=true`.
+
```
Comment (gp201) [inline Webpack.md]:
``````suggestion
Docker:
```
make run-devserver prod_env=true`.
```

``````

File: Webpack.md [hunk 4]
```diff
@@ -151,8 +151,13 @@ output: {
 Here we specify the build mode i.e. production and the output config. We don't use a devtool here to make the build faster.
 
 This is the config that is used while running the dev server using
+
+Python:
 `python -m scripts.start --prod_env`.
```
Comment (gp201) [inline Webpack.md]: Ditto

File: Debug-frontend-code.md [hunk 5]
```diff
@@ -60,11 +60,18 @@ If you select `webpack://` in the sidebar of the sources tab, you can see Oppia'
 
 However, here you are seeing the code that has been altered by the build process. To see the original source code, you need to enable source maps when running the development server like this:
 
+Pytho:
```
Comment (gp201) [inline Debug-frontend-code.md]:
```suggestion
Python:
```

**Review by gp201 — CHANGES_REQUESTED** [review]
Thanks, @jaiakash ! I left some comments. PTAL
Please go ahead and update the remaining pages

File: Acceptance-Tests.md [hunk 6]
```diff
@@ -46,10 +46,21 @@ Files included inside this directory are :
...
+
+Docker:
+```
+make run_tests.acceptance suite=SUITE_NAME
+```
 
 For example, to run the `check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js` test, run the following command:
-`python -m scripts.run_acceptance_tests --suite="blog-editor-tests/check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js"`
+Python:
+```python -m scripts.run_acceptance_tests --suite="blog-editor-tests/check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js"```
```
Comment (gp201) [inline Acceptance-Tests.md]:
``````suggestion
```
python -m scripts.run_acceptance_tests --suite="blog-editor-tests/check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js"
```
``````

File: Acceptance-Tests.md [hunk 7]
```diff
@@ -46,10 +46,21 @@ Files included inside this directory are :
...
+
+Docker:
+```make run_tests.acceptance suite="blog-editor-tests/check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js"```
```
Comment (gp201) [inline Acceptance-Tests.md]:
``````suggestion
```
make run_tests.acceptance suite="blog-editor-tests/check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js"
```
``````

File: Backend-Type-Annotations.md [hunk 8]
```diff
@@ -18,9 +18,18 @@ Mypy checks script (`scripts/run_mypy_checks.py`) is the script used to run our
 
 It has two modes of running:
 
-1. `python -m scripts.run_mypy_checks`\
+1. Python:
+  `python -m scripts.run_mypy_checks`\
+
+   Docker:
+   `make run_tests.mypy`
```
Comment (gp201) [inline Backend-Type-Annotations.md]: Write all commands in triple backticks (```)

**Review by gp201 — CHANGES_REQUESTED** [review]
I left a few comments. Have you tested all the commands?

File: Backend-Type-Annotations.md (hunk 8 above)
Comment (jaiakash) [reply Backend-Type-Annotations.md]: ok

**Comment by jaiakash** [comment]
Please cross-check the commands @yash1378.

And PTAL @gp201

**Comment by 0xV0YD** [comment]
@jaiakash i have checked the commands also and they are working just in 1  command i was having some errors and had doubts about it so i have asked @gp201  about it on google chat and he'll tell about whether to ignore that error as of now of is it actually worth considering right now ,  this is that error 
Screenshot from 2024-02-01 05-03-43

**Comment by 0xV0YD** [comment]
PTAL @gp201

File: Build-process.md [hunk 9]
```diff
@@ -119,9 +120,16 @@ Even when running in dev mode, the code your browser runs does not look much lik
...
+Python:
 ```console
 python -m scripts.start --source_maps
 python -m scripts.run_e2e_tests --prod_env --source_maps
 ```
 
+Docker:
+```console
+make run-devserver source_maps=true
+make run_e2e_tests.prod prod_env=true source_maps=true
```
Comment (gp201) [inline Build-process.md]: incorrect command please fix

File: Debug-frontend-code.md [hunk 10]
```diff
@@ -60,11 +60,18 @@ If you select `webpack://` in the sidebar of the sources tab, you can see Oppia'
...
+Python:
 ```console
 python -m scripts.start --source_maps
 python -m scripts.run_e2e_tests --prod_env --source_maps
 ```
 
+Docker:
+```console
+make run-devserver source_maps=true
+make run_e2e_tests.prod prod_env=true source_maps=true
```
Comment (gp201) [inline Debug-frontend-code.md]: incorrect command please fix

**Review by gp201 — CHANGES_REQUESTED** [review]

**Comment by gp201** [comment]
@jaiakash PTAL

**Comment by jaiakash** [comment]
PTAL the new command @yash1378 @gp201

**Comment by yash1378** [comment]
@jaiakash   actually rest all commands are working well for me but the commands majorly which rrquire the running of devserver , in those commands generally i am getting one particular error

**Comment by 0xV0YD** [comment]
@gp201 @jaiakash  commands working fine for me , as after resolving the error which i was facing remaining commands also working fine for me

**Comment by 0xV0YD** [comment]
PTAL @gp201

File: Build-process.md [hunk 11]
```diff
@@ -84,6 +84,7 @@ First, let's consider what happens in plain prod mode where those two options ar
 
 This is the mode our tests run in (except the lighthouse accessibility tests), and it's what you'll want to use if you want to locally run a version of the app that's as close to the production version as possible. You can use this mode like this:
 
+Python:
 ```console
 python -m scripts.start --prod_env
 ```
```
Comment (gp201) [inline Build-process.md:90]: What is the Docker cmd for this?

File: How-to-develop-for-i18n.md [hunk 12]
```diff
@@ -46,10 +46,16 @@ If a translation key is not used any more, you must delete it from `assets/i18n/
 
 To verify your changes to translation files locally, run the following command in a terminal:
 
+Python:
 ```
     python -m scripts.run_backend_tests --test_target=core.controllers.base_test.I18nDictsTests
```
Comment (gp201) [inline How-to-develop-for-i18n.md]:
```suggestion
python -m scripts.run_backend_tests --test_target=core.controllers.base_test.I18nDictsTests
```

File: How-to-develop-for-i18n.md [hunk 13]
```diff
@@ -46,10 +46,16 @@ If a translation key is not used any more, you must delete it from `assets/i18n/
...
 ```
 
+Docker:
+```
+    make run_backend_tests PYTHON_ARGS="--test_target=core.controllers.base_test.I18nDictsTests"
```
Comment (gp201) [inline How-to-develop-for-i18n.md]:
```suggestion
make run_backend_tests PYTHON_ARGS="--test_target=core.controllers.base_test.I18nDictsTests"
```

**Review by gp201 — COMMENTED** [review]
Thanks, @jaiakash and @yash1378 ! I left some comments. PTAL

File: Build-process.md (hunk 11 above)
Comment (jaiakash) [reply Build-process.md:90]: Done

**Comment by jaiakash** [comment]
Did the changes, PTAL @yash1378 @gp201

File: Build-process.md [hunk 14]
```diff
@@ -89,16 +89,27 @@ Python:
...
 
+Docker:
+```console
+make run-devserver PYTHON_ARGS="--prod_env"
```
Comment (gp201) [inline Build-process.md]:
```suggestion
make run-devserver prod_env=true
```

File: Build-process.md [hunk 15]
```diff
@@ -89,16 +89,27 @@ Python:
...
 This mode is used by our deployment scripts. You can also enable it locally like this:
 
+Python:
 ```console
 python -m scripts.start --prod_env --maintenance_mode
 ```
 
+Docker:
+```console
+make run-devserver PYTHON_ARGS="--prod_env --maintenance_mode"
```
Comment (gp201) [inline Build-process.md]: Please refer to for the commands: https://github.com/oppia/oppia/wiki/Installing-Oppia-using-Docker#using-flags-with-make-command

**Review by gp201 — CHANGES_REQUESTED** [review]
@jaiakash PTAL

**Comment by jaiakash** [comment]
> @jaiakash PTAL

My bad, for ```source_maps``` ,  ```maintenance_mode``` and ```prod_env``` , there is an option to pass arguments directly without wrapping it with ```PYTHON_ARGS```. 

Did the changes, PTAL @yash1378 @gp201

File: End-to-End-Tests.md [hunk 16]
```diff
@@ -258,7 +258,7 @@ python -m scripts.run_e2e_tests --prod_env
 
 Docker:
 ```console
-make run_tests.e2e PYTHON_ARGS="--prod_env"
+make run_tests.e2e prod_env=true
```
Comment (gp201) [inline End-to-End-Tests.md]: Please undo this change. Refer to https://github.com/oppia/oppia/wiki/Installing-Oppia-using-Docker#running-tests

File: Debug-frontend-code.md [hunk 17]
```diff
@@ -68,8 +68,8 @@ python -m scripts.run_e2e_tests --prod_env --source_maps
 
 Docker:
 ```console
-make run-devserver PYTHON_ARGS="--source_maps"
-make run_tests.e2e PYTHON_ARGS="--prod_env --source_maps"
+make run-devserver source_maps=true
+make run_tests.e2e prod_env=true source_maps=true
```
Comment (gp201) [inline Debug-frontend-code.md]: Please undo this change. Refer to https://github.com/oppia/oppia/wiki/Installing-Oppia-using-Docker#running-tests

File: Build-process.md (hunk 15 above)
Comment (gp201) [reply Build-process.md]: Please fix this too. Thanks!

**Review by gp201 — CHANGES_REQUESTED** [review]
@jaiakash PTAL

File: End-to-End-Tests.md (hunk 16 above)
Comment (jaiakash) [reply End-to-End-Tests.md]: Done

File: Debug-frontend-code.md (hunk 17 above)
Comment (jaiakash) [reply Debug-frontend-code.md]: Done

**Comment by jaiakash** [comment]
Done @gp201

**Comment by gp201** [comment]
@jaiakash Please address all comments and assign me. Thanks!
https://github.com/oppia/oppia-web-developer-docs/pull/244#discussion_r1482184022

File: Angular-Migration.md [hunk 18]
```diff
@@ -751,23 +751,41 @@ There may be some style updates required to make sure that the pages look exactl
...
 1. Ensure your frontend tests pass
 
+   Python:
    ```console
    python -m scripts.run_frontend_tests
    ```
+   
+   Docker:
+   ```console
+   make run_tests.frontend
```
Comment (0xV0YD) [inline Angular-Migration.md:761]: this command is getting 8885 success from 8886  and throwing this error below in picture 
Screenshot from 2024-02-08 18-04-50

Screenshot from 2024-02-08 18-07-26

File: Angular-Migration.md [hunk 19]
```diff
@@ -751,23 +751,41 @@ There may be some style updates required to make sure that the pages look exactl
...
 2. Ensure there are no typescript errors:
 
+   Python:
    ```console
    python -m scripts.typescript_checks
    ```
 
+    Docker:
+    ```console
+    make run_tests.typescript
```
Comment (0xV0YD) [inline Angular-Migration.md:775]: successful

File: Acceptance-Tests.md [hunk 20]
```diff
@@ -46,10 +46,25 @@ Files included inside this directory are :
...
 For example, to run the `check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js` test, run the following command:
-`python -m scripts.run_acceptance_tests --suite="blog-editor-tests/check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js"`
+Python:
+```
+python -m scripts.run_acceptance_tests --suite="blog-editor-tests/check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js"
+```
+
+Docker:
+```
+make run_tests.acceptance suite="blog-editor-tests/check-blog-editor-unable-to-publish-duplicate-blog-post.spec.js"
```
Comment (0xV0YD) [inline Acceptance-Tests.md:66]: successful

File: Acceptance-Tests.md [hunk 21]
```diff
@@ -46,10 +46,25 @@ Files included inside this directory are :
...
 ## How to run the acceptance tests
 From the root directory of oppia, run the following command:
-`python -m scripts.run_acceptance_tests --suite={{suiteName}}`
+```  
+python -m scripts.run_acceptance_tests --suite={{suiteName}}  
+``` 
+
+Docker:
+```
+make run_tests.acceptance suite=SUITE_NAME
```
Comment (0xV0YD) [inline Acceptance-Tests.md:55]: successful

File: Angular-Migration.md [hunk 22]
```diff
@@ -751,23 +751,41 @@ There may be some style updates required to make sure that the pages look exactl
...
 3. Ensure there are no linting errors:
 
+   Python:
    ```console
    python -m scripts.linters.pre_commit_linter
    ```
+   
+    Docker:
+    ```console
+    make run_tests.lint
```
Comment (0xV0YD) [inline Angular-Migration.md:787]: successful

File: Backend-Type-Annotations.md [hunk 23]
```diff
@@ -18,9 +18,26 @@ Mypy checks script (`scripts/run_mypy_checks.py`) is the script used to run our
...
 
-1. `python -m scripts.run_mypy_checks`\
+1. Python:
+```
+python -m scripts.run_mypy_checks
+```
+
+   Docker:
+```
+make run_tests.mypy
```
Comment (0xV0YD) [inline Backend-Type-Annotations.md:28]: successful

File: Backend-Type-Annotations.md [hunk 24]
```diff
@@ -18,9 +18,26 @@ Mypy checks script (`scripts/run_mypy_checks.py`) is the script used to run our
...
   This runs the type checks on all the type annotated files in the codebase.
-2. `python -m scripts.run_mypy_checks --files path/file1.py path/file2.py`\
+2. Python:
+```
+python -m scripts.run_mypy_checks --files path/file1.py path/file2.py
+```
+
+   Docker:
+```
+make run_tests.mypy PYTHON_ARGS="--files path/file1.py path/file2.py"
```
Comment (0xV0YD) [inline Backend-Type-Annotations.md:39]: successful

File: Backend-tests.md [hunk 25]
```diff
@@ -151,33 +151,57 @@ In fact, too many mocks can be a problem because when someone changes the code y
...
 You can run backend tests like this:
 
+Python:
 ```console
 python -m scripts.run_backend_tests
 ```
 
+Docker:
+```console
+make run_tests.backend
```
Comment (0xV0YD) [inline Backend-tests.md:161]: this command is throwing an error in as speicified in below image but that will come as docker setup is in building progress as said by praneeth 
Screenshot from 2024-02-08 18-11-21

File: Backend-tests.md [hunk 26]
```diff
@@ -151,33 +151,57 @@ In fact, too many mocks can be a problem because when someone changes the code y
...
 Alternatively, you can run just a single test module like this:
 
+Python:
 ```console
 python -m scripts.run_backend_tests --test_target=core.controllers.editor_test
 ```
 
+Docker:
+```console
+make run_tests.backend PYTHON_ARGS="--test_target=core.controllers.editor_test"
```
Comment (0xV0YD) [inline Backend-tests.md:173]: this is also throwing 
Screenshot from 2024-02-08 18-11-21

Comment (0xV0YD) [inline Backend-tests.md:173]: this is also throwing 
Screenshot from 2024-02-08 18-11-21

File: Backend-tests.md [hunk 27]
```diff
@@ -151,33 +151,57 @@ In fact, too many mocks can be a problem because when someone changes the code y
...
 If you also want to see the output of `print` statements and error logs in the terminal, use `--verbose` like this:
 
+Python:
 ```console
 python -m scripts.run_backend_tests --test_target=core.controllers.editor_test --verbose
 ```
 
+Docker:
+```console
+make run_tests.backend PYTHON_ARGS="--test_target=core.controllers.editor_test --verbose"
```
Comment (0xV0YD) [inline Backend-tests.md:190]: this is also thworing same 
Screenshot from 2024-02-08 18-11-21

File: Backend-tests.md [hunk 28]
```diff
@@ -151,33 +151,57 @@ In fact, too many mocks can be a problem because when someone changes the code y
...
 For more information about `--test_target` and other flags, run:
 
+Python:
 ```console
 python -m scripts.run_backend_tests --help
 ```
 
+Docker:
+```console
+make run_tests.backend PYTHON_ARGS="--help"
```
Comment (0xV0YD) [inline Backend-tests.md:202]: successful

File: Backend-tests.md [hunk 29]
```diff
@@ -209,10 +233,16 @@ We use a simple tool called *code coverage* to check that all of Oppia’s backe
...
 When writing a test for a function or class, you can generate a coverage report to verify that all the lines of the function/class have been included in the tests. To do this, simply add the `--generate_coverage_report` flag to the `run_backend_tests` command:
 
+Python:
 ```console
 python -m scripts.run_backend_tests --generate_coverage_report
 ```
 
+Docker:
+```console
+make run_tests.backend PYTHON_ARGS="--generate_coverage_report"
```
Comment (0xV0YD) [inline Backend-tests.md:243]: same error 
Screenshot from 2024-02-08 18-11-21

File: Build-process.md [hunk 30]
```diff
@@ -84,20 +84,32 @@ First, let's consider what happens in plain prod mode where those two options ar
...
 This is the mode our tests run in (except the lighthouse accessibility tests), and it's what you'll want to use if you want to locally run a version of the app that's as close to the production version as possible. You can use this mode like this:
 
+Python:
 ```console
 python -m scripts.start --prod_env
 ```
 
+Docker:
+```console
+make run-devserver prod_env=true
```
Comment (0xV0YD) [inline Build-process.md:94]: successful

File: Build-process.md [hunk 31]
```diff
@@ -84,20 +84,32 @@ First, let's consider what happens in plain prod mode where those two options ar
...
 This mode is used by our deployment scripts. You can also enable it locally like this:
 
+Python:
 ```console
 python -m scripts.start --prod_env --maintenance_mode
 ```
 
+Docker:
+```console
+make run-devserver prod_env=true maintenance_mode=true
```
Comment (0xV0YD) [inline Build-process.md:110]: successful but showing this 
Screenshot from 2024-02-08 21-35-15

File: Build-process.md [hunk 32]
```diff
@@ -119,9 +131,16 @@ Even when running in dev mode, the code your browser runs does not look much lik
...
 
+Python:
 ```console
 python -m scripts.start --source_maps
 python -m scripts.run_e2e_tests --prod_env --source_maps
 ```
 
+Docker:
+```console
+make run-devserver source_maps=true
```
Comment (0xV0YD) [inline Build-process.md:142]: successful

File: Build-process.md [hunk 33]
```diff
@@ -119,9 +131,16 @@ Even when running in dev mode, the code your browser runs does not look much lik
...
+make run_tests.e2e prod_env=true source_maps=true
```
Comment (0xV0YD) [inline Build-process.md:143]: successful

File: Custom-Pylint-checks.md [hunk 34]
```diff
@@ -527,8 +527,14 @@ def test_finds_hello_world_assignment(self):
...
 Once you've added test cases, you can run them like this:
 
+Python:
 ```console
 python -m scripts.run_backend_tests --test_target=scripts.linters.pylint_extensions_test --verbose
 ```
 
+Docker:
+```console
+make run_backend_tests PYTHON_ARGS="--test_target=scripts.linters.pylint_extensions_test --verbose"
```
Comment (0xV0YD) [inline Custom-Pylint-checks.md]: Screenshot from 2024-02-08 21-37-29
this error is coming

File: Debug-backend-tests.md [hunk 35]
```diff
@@ -72,31 +72,55 @@ When you know which test is causing you problems, running it in isolation can he
...
 To run a test in isolation, you can use the `--test_target` option:
 
+Python:
 ```console
 python -m scripts.run_backend_tests --test_target jobs.jobs_manager_test.RefreshStateOfBeamJobRunModelTests.test_failed_api_call_logs_the_exception
 ```
 
+Docker:
+```console
+make run_tests.backend PYTHON_ARGS="--test_target jobs.jobs_manager_test.RefreshStateOfBeamJobRunModelTests.test_failed_api_call_logs_the_exception"
```
Comment (0xV0YD) [inline Debug-backend-tests.md:82]: Screenshot from 2024-02-08 21-39-22
again same this module error is coming due to which the make stop is getting activated

File: Debug-backend-tests.md [hunk 36]
```diff
@@ -72,31 +72,55 @@ When you know which test is causing you problems, running it in isolation can he
...
 If you wanted to run all the tests defined by the `RefreshStateOfBeamJobRunModelTests`, you could do that too. Just shorten the dotted name to end at the class:
 
+Python:
 ```console
 python -m scripts.run_backend_tests --test_target jobs.jobs_manager_test.RefreshStateOfBeamJobRunModelTests
 ```
 
+Docker:
+```console
+make run_tests.backend PYTHON_ARGS="--test_target jobs.jobs_manager_test.RefreshStateOfBeamJobRunModelTests"
```
Comment (yash1378) [inline Debug-backend-tests.md:94]: same error

Comment (0xV0YD) [inline Debug-backend-tests.md:94]: Screenshot from 2024-02-08 21-39-22
same error

File: Debug-backend-tests.md [hunk 37]
```diff
@@ -72,31 +72,55 @@ When you know which test is causing you problems, running it in isolation can he
...
 Note that if you want to run all the tests in a directory, you need to use `--test_path` instead like this:
 
+Python:
 ```console
 python -m scripts.run_backend_tests --test_path jobs
 ```
 
+Docker:
+```console
+make run_tests.backend PYTHON_ARGS="--test_path jobs"
```
Comment (0xV0YD) [inline Debug-backend-tests.md:106]: Screenshot from 2024-02-08 21-42-23
this error i think could be in my setup only as due to some mistake the file could be missing so update me please is this file exists in your codebase

File: Debug-backend-tests.md [hunk 38]
```diff
@@ -72,31 +72,55 @@ When you know which test is causing you problems, running it in isolation can he
...
 
 To identify the missing coverage lines run the below command.
+Python:
 ```console
 python -m scripts.run_backend_tests --generate_coverage_report
 ```
 
+Docker:
+```console
+make run_tests.backend PYTHON_ARGS="--generate_coverage_report"
```
Comment (0xV0YD) [inline Debug-backend-tests.md:121]: same that module dev_server not found error

File: Debug-end-to-end-tests.md [hunk 39]
```diff
@@ -38,10 +38,16 @@ There are many ways to go about debugging an E2E test, but here is one approach:
...
 2. Run the e2e script with the flag `--debug_mode`. For example,
 
+   Python:
    ```console
    python -m scripts.run_e2e_tests --debug_mode --suite="topicAndStoryEditor"
    ```
 
+   Docker:
+   ```console
+   make run_tests.e2e PYTHON_ARGS="--debug_mode --suite="topicAndStoryEditor"
```
Comment (0xV0YD) [inline Debug-end-to-end-tests.md:48]: successful

File: Debug-frontend-code.md [hunk 40]
```diff
@@ -60,11 +60,18 @@ If you select `webpack://` in the sidebar of the sources tab, you can see Oppia'
...
 
+Python:
 ```console
 python -m scripts.start --source_maps
 python -m scripts.run_e2e_tests --prod_env --source_maps
 ```
 
+Docker:
+```console
+make run-devserver source_maps=true
```
Comment (0xV0YD) [inline Debug-frontend-code.md:71]: success

File: Debug-frontend-code.md [hunk 41]
```diff
@@ -60,11 +60,18 @@ If you select `webpack://` in the sidebar of the sources tab, you can see Oppia'
...
+make run_tests.e2e PYTHON_ARGS="--prod_env --source_maps"
```
Comment (0xV0YD) [inline Debug-frontend-code.md:72]: success

File: Debug-frontend-tests.md [hunk 42]
```diff
@@ -88,10 +88,16 @@ The tests should work correctly regardless of what order they run in. If you fin
...
 By default, we suppress log output from the tests to the terminal when running the frontend tests. This keeps the test output clean so it's easy to find which tests failed. However, this output can be useful for debugging. For example, if you add `console.log()` or `console.error()` statements to your tests for debugging, the output from those statements will be suppressed by default. To view this output, pass `--verbose` when you run the frontend tests:
 
+Python:
 ```console
 python -m scripts.run_frontend_tests --verbose
 ```
 
+Docker:
+```console
+make run_tests.frontend PYTHON_ARGS="--verbose"
```
Comment (0xV0YD) [inline Debug-frontend-tests.md:98]: success

File: Debug-frontend-tests.md [hunk 43]
```diff
@@ -103,10 +109,16 @@ Note that depending on your situation, `console.log()` or `console.error()` migh
...
 By default, the combined-tests.spec.js file is deleted after running the frontend tests. This is because it is a large file that is a combination of all the frontend test files. Currently, the stack traces for frontend test failures mention the line numbers with respect to the combined-tests.spec.js file. To allow you to easily find the line in the combined-tests.spec.js file, you can download the file and view it in a text editor using the `--download_combined_frontend_spec_file` flag:
 
+Python:
 ```console
 python -m scripts.run_frontend_tests --download_combined_frontend_spec_file
 ```
 
+Docker:
+```console
+make run_tests.frontend PYTHON_ARGS="--download_combined_frontend_spec_file"
```
Comment (0xV0YD) [inline Debug-frontend-tests.md:119]: success

File: End-to-End-Tests.md [hunk 44]
```diff
@@ -167,10 +167,16 @@ Extensions provide `webdriverio.js` files to make them easier to test. The E2E t
...
 If you don't know the name of the suite you want to run, you can find it in `core/tests/wdio.conf.js`. Then you can run your test like this:
 
+Python:
 ```console
 python -m scripts.run_e2e_tests --suite="suiteName"
 ```
 
+Docker:
+```console
+make run_tests.e2e PYTHON_ARGS="--suite=suiteName"
```
Comment (0xV0YD) [inline End-to-End-Tests.md:177]: throwing error  but also giving all successes 
Screenshot from 2024-02-08 21-55-58

File: End-to-End-Tests.md [hunk 45]
```diff
@@ -185,10 +191,16 @@ You may also want to set the chromedriver version manually if you want to test a
...
 To manually set the chromedriver version, use the `--chrome_driver_version` argument:
 
+Python:
 ```console
 python -m scripts.run_e2e_tests --chrome_driver_version <version>
 ```
 
+Docker:
+```console
+make run_tests.e2e PYTHON_ARGS="--chrome_driver_version <version>"
```
Comment (0xV0YD) [inline End-to-End-Tests.md:201]: success

File: End-to-End-Tests.md [hunk 46]
```diff
@@ -207,16 +219,28 @@ If you see a failure due to the webdriver, please follow the instructions above
...
 If you run all the E2E tests at once (i.e. if you don't specify a suite), the tests will be sharded across multiple Chrome browser instances. By default, the tests will use 3 shards (i.e. 3 browsers). If you do this, you should close background processes to maximize the compute resources available to the tests. You can configure the number of shards like this:
 
+Python:
 ```console
 python -m scripts.run_e2e_tests --sharding-instances=<number of shards>
 ```
 
+Docker:
+```console
+make run_tests.e2e PYTHON_ARGS="--sharding-instances=<number of shards>"
```
Comment (0xV0YD) [inline End-to-End-Tests.md:229]: success

File: End-to-End-Tests.md [hunk 47]
```diff
@@ -207,16 +219,28 @@ If you see a failure due to the webdriver, please follow the instructions above
...
 You can disable sharding like this:
 
+Python:
 ```console
 python -m scripts.run_e2e_tests --sharding=false
 ```
 
+Docker:
+```console
+make run_tests.e2e PYTHON_ARGS="--sharding=false"
```
Comment (0xV0YD) [inline End-to-End-Tests.md:241]: success

File: End-to-End-Tests.md [hunk 48]
```diff
@@ -227,10 +251,16 @@ To run just one test, change the "it" to "fit" for that test. Then when you run
...
 To run the end-to-end tests in production mode, use the `--prod_env` flag:
 
+Python:
 ```console
 python -m scripts.run_e2e_tests --prod_env
 ```
 
+Docker:
+```console
+make run_tests.e2e PYTHON_ARGS="--prod_env"
```
Comment (0xV0YD) [inline End-to-End-Tests.md:261]: success

File: How-to-develop-for-i18n.md [hunk 49]
```diff
@@ -46,8 +46,14 @@ If a translation key is not used any more, you must delete it from `assets/i18n/
...
 
+Python:
 ```
-    python -m scripts.run_backend_tests --test_target=core.controllers.base_test.I18nDictsTests
+python -m scripts.run_backend_tests --test_target=core.controllers.base_test.I18nDictsTests
+```
+
+Docker:
+```
+make run_backend_tests PYTHON_ARGS="--test_target=core.controllers.base_test.I18nDictsTests"
```
Comment (0xV0YD) [inline How-to-develop-for-i18n.md]: throwing error

File: TypeScript-tests.md [hunk 50]
```diff
@@ -13,10 +13,16 @@ For help writing TypeScript code, check out our [[guide on defining types|Guide-
...
 You can run the TypeScript tests like this:
 
+Python:
 ```console
 python -m scripts.typescript_checks
 ```
 
+Docker:
+```console
+make run_tests.typescript
```
Comment (0xV0YD) [inline TypeScript-tests.md:23]: success

File: Webpack.md [hunk 51]
```diff
@@ -151,7 +151,16 @@ output: {
...
-`python -m scripts.start --prod_env`.
+
+Python:
+```
+python -m scripts.start --prod_env
+```
+
+Docker:
+```
+make run-devserver prod_env=true
```
Comment (0xV0YD) [inline Webpack.md:162]: success

File: End-to-End-Tests.md (hunk 44 above)
Comment (gp201) [reply End-to-End-Tests.md:177]: Try this https://stackoverflow.com/a/32600959
It should fix the issue

File: Custom-Pylint-checks.md (hunk 34 above)
Comment (gp201) [reply Custom-Pylint-checks.md]: @jaiakash Please fix this error.

File: Backend-tests.md (hunk 25 above)
Comment (gp201) [reply Backend-tests.md:161]: This error from backend tests can be ignored for now.

File: How-to-develop-for-i18n.md (hunk 49 above)
Comment (gp201) [reply How-to-develop-for-i18n.md]: @jaiakash Please fix this cmd

**Comment by gp201** [comment]
@jaiakash please fix the following:
https://github.com/oppia/oppia-web-developer-docs/pull/244#discussion_r1483420852
https://github.com/oppia/oppia-web-developer-docs/pull/244#discussion_r1483417937

@yash1378 please try the following
https://github.com/oppia/oppia-web-developer-docs/pull/244#discussion_r1483417054

File: End-to-End-Tests.md (hunk 44 above)
Comment (0xV0YD) [reply End-to-End-Tests.md:177]: > Try this https://stackoverflow.com/a/32600959 It should fix the issue

yeah after changing the max user watches working fine

File: How-to-develop-for-i18n.md (hunk 49 above)
Comment (0xV0YD) [reply How-to-develop-for-i18n.md]: i guess this command has some naming issues 
Screenshot from 2024-02-10 13-27-06
as you can see run_backend_tests is not there instead that is run_tests.backend

Comment (0xV0YD) [reply How-to-develop-for-i18n.md]: also the error is also stating that 
Screenshot from 2024-02-10 13-31-53

Comment (0xV0YD) [reply How-to-develop-for-i18n.md]: @jaiakash @gp201  this is resolved

File: Custom-Pylint-checks.md (hunk 34 above)
Comment (0xV0YD) [reply Custom-Pylint-checks.md]: also corrected the command here also @gp201 but still its throwing some error as shown in image below 
Screenshot from 2024-02-10 14-51-16

**Review by gp201 — APPROVED** [review]
LGTM

File: Angular-Migration.md (hunk 18 above)
Comment (0xV0YD) [reply Angular-Migration.md:761]: what about this @gp201 ? is this ok should be reviewed ? or this will also be resolved by that stackoverflow solution which you mentioned as the error is same i guess ?
